devito/dse/backends/basic.py

```diff
@@ -0,0 +1,45 @@
+from sympy import cos, sin
+
+from devito.dse.backends import AbstractRewriter, dse_pass
+from devito.dse.extended_sympy import bhaskara_cos, bhaskara_sin
+from devito.dse.manipulation import common_subexprs_elimination
+
+from devito.interfaces import ScalarFunction
+
+
+class BasicRewriter(AbstractRewriter):
+
+    def _pipeline(self, state):
+        self._eliminate_intra_stencil_redundancies(state)
+
+    @dse_pass
+    def _eliminate_intra_stencil_redundancies(self, cluster, **kwargs):
+        """
+        Perform common subexpression elimination, bypassing the scalar expressions
+        extracted in previous passes.
+        """
+
+        skip = [e for e in cluster.exprs if e.lhs.base.function.is_SymbolicFunction]
```

Comment:
This is me after comments in another PRs: we were starting having list comprehensions with `i`, `x`, `j` more or less at random, so I was trying to have `k, v` for dicts, `i` for lists, etc. 

In the DSE it's kinda special, because for all list comprehensions involving SymPy expressions I've used `e`

as I pointed out already, this was just me nitpicking